pkg/appmanager/appManager.go

```diff
@@ -90,20 +89,30 @@ type Manager struct {
 	broadcaster   record.EventBroadcaster
 	eventRecorder record.EventRecorder
 	eventSource   v1.EventSource
+	// Route configurations
+	routeConfig RouteConfig
 }
 
 // Struct to allow NewManager to receive all or only specific parameters.
 type Params struct {
 	KubeClient      kubernetes.Interface
 	restClient      rest.Interface // package local for unit testing only
-	RouteClientV1   *routeclient.Client
+	RouteClientV1   rest.Interface
```

Comment:
This may cause difficulty down the road when we need to send events on a Route object. As I recall, the routeclient.Client was an object whose only member was the rest.Interface, but it also satisfied the interface for sending updates back to k8s.